utils/sshclient.go

```diff
@@ -0,0 +1,383 @@
+				"arcfour",
+			},
+		},
+		HostKeyCallback: ssh.HostKeyCallback(func(hostname string, remote net.Addr, key ssh.PublicKey) error { return nil }),
+	}
+
+	return Dial("tcp", addr, config)
+}
+
+// DialWithKey starts a client connection to the given SSH server with key authmethod.
+func DialWithKey(addr, user, keyfile string) (*Client, error) {
+	key, err := ioutil.ReadFile(keyfile)
+	if err != nil {
+		return nil, err
+	}
+
+	signer, err := ssh.ParsePrivateKey(key)
+	if err != nil {
+		return nil, err
+	}
+
+	config := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		HostKeyCallback: ssh.HostKeyCallback(func(hostname string, remote net.Addr, key ssh.PublicKey) error { return nil }),
+	}
+
+	return Dial("tcp", addr, config)
+}
+
+// DialWithKeyWithPassphrase same as DialWithKey but with a passphrase to decrypt the private key
+func DialWithKeyWithPassphrase(addr, user, keyfile string, passphrase string) (*Client, error) {
+	key, err := ioutil.ReadFile(keyfile)
+	if err != nil {
+		return nil, err
+	}
+
+	signer, err := ssh.ParsePrivateKeyWithPassphrase(key, []byte(passphrase))
+	if err != nil {
+		return nil, err
+	}
+
+	config := &ssh.ClientConfig{
+		User: user,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+		HostKeyCallback: ssh.HostKeyCallback(func(hostname string, remote net.Addr, key ssh.PublicKey) error { return nil }),
+	}
+
+	return Dial("tcp", addr, config)
+}
+
+// Dial starts a client connection to the given SSH server.
+// This wraps ssh.Dial.
+func Dial(network, addr string, config *ssh.ClientConfig) (*Client, error) {
+	client, err := ssh.Dial(network, addr, config)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{
+		client: client,
+	}, nil
+}
+
+// Close closes the underlying client network connection.
+func (c *Client) Close() error {
+	return c.client.Close()
+}
+
+// UnderlyingClient get the underlying client.
+func (c *Client) UnderlyingClient() *ssh.Client {
+	return c.client
+}
+
+// Cmd creates a RemoteScript that can run the command on the client. The cmd string is split on newlines and each line is executed separately.
+func (c *Client) Cmd(cmd string) *RemoteScript {
+	return &RemoteScript{
+		_type:  cmdLine,
+		client: c.client,
+		script: bytes.NewBufferString(cmd + "\n"),
+	}
+}
+
+// Script creates a RemoteScript that can run the script on the client.
+func (c *Client) Script(script string) *RemoteScript {
+	return &RemoteScript{
+		_type:  rawScript,
+		client: c.client,
+		script: bytes.NewBufferString(script + "\n"),
+	}
+}
+
+// ScriptFile creates a RemoteScript that can read a local script file and run it remotely on the client.
+func (c *Client) ScriptFile(fname string) *RemoteScript {
+	return &RemoteScript{
+		_type:      scriptFile,
+		client:     c.client,
+		scriptFile: fname,
+	}
+}
+
+// A RemoteScript represents script that can be run remotely.
+type RemoteScript struct {
+	client     *ssh.Client
+	_type      remoteScriptType
```

Comment:
can u help to change the field name? like scriptType

Reply:
I see.